src/main/java/com/hazelcast/remotecontroller/HazelcastCloudManager.java

```diff
@@ -0,0 +1,336 @@
+                String responseBody = prepareAndSendRequest(query).body().string();
+                LOG.info(responseBody);
+                currentState = mapper.readTree(responseBody).get("data").get("cluster").get("state").asText();
+                if(currentState.equalsIgnoreCase(expectedState))
+                    return;
+                if(currentState.equalsIgnoreCase("FAILED"))
+                {
+                    throw new CloudException("Cluster is failed");
+                }
+                if((System.currentTimeMillis() - startTime) + TimeUnit.SECONDS.toMillis(retryTimeInSecond)  > timeoutInMillisecond)
+                    break;
+                TimeUnit.SECONDS.sleep(retryTimeInSecond);
+            }
+            throw new CloudException(String.format("The cluster with id %s could not come to the given state in %d millisecond", clusterId, timeoutInMillisecond));
+        } catch (Exception e) {
+            throw new CloudException(String.format("Wait for state of cluster with id %s failed. Rc stack trace is: %s", clusterId, e.getStackTrace()));
+        }
+    }
+
+    private void waitForDeletedCluster(String clusterId, long timeoutInMillisecond) throws CloudException, InterruptedException {
+        long startTime = System.currentTimeMillis();
+        while (true) {
+            if(getHazelcastCloudCluster(clusterId) == null)
+                return;
+            if((System.currentTimeMillis() - startTime) + TimeUnit.SECONDS.toMillis(retryTimeInSecond)  > timeoutInMillisecond)
+                break;
+            TimeUnit.SECONDS.sleep(retryTimeInSecond);
+        }
+        throw new CloudException(String.format("The cluster with id %s is not deleted in %d millisecond", clusterId, timeoutInMillisecond));
+    }
+
+    private String getBearerToken(String apiKey, String apiSecret) throws CloudException {
+        try {
+            String query = String.format("{ \"query\": \"mutation { login(apiKey: \\\"%s\\\", apiSecret: \\\"%s\\\") {token} }\" }", apiKey, apiSecret);
+            RequestBody body = RequestBody.create(JSON, query);
+            Request request = new Request.Builder()
+                    .url(HttpUrl.get(uri))
+                    .post(body)
+                    .header("Content-Type", "application/json")
+                    .build();
+            call = client.newCall(request);
+            Response response = call.execute();
+            String loginResponse = response.body().string();
+            LOG.info(maskValueOfToken(loginResponse));
+            return mapper.readTree(loginResponse).get("data").get("login").get("token").asText();
+        } catch (Exception e) {
+            throw new CloudException("Get bearer token is failed");
+        }
+    }
+
+    // It is not possible to connect a cluster with its actual name, it seems a bug of the cloud API, that method is a workaround.
+    // There could be a better way to get this name, TODO: contact with cloud team.
```

Comment:
why not open an issue at cloud repo?

Reply:
They are aware of it and AFAIK there is an issue on their repo.